paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/refund/PaytrailRefundPaymentService.java

```diff
@@ -245,47 +244,44 @@ private RefundPayment createRefundPayment(
         return refundPayment;
     }
 
-    public void triggerRefundPaymentPaidEvent(RefundPayment payment) {
+    public void triggerRefundPaymentPaidEvent(RefundPayment refundPayment) {
         String now = DateTimeUtil.getDateTime();
 
-        PaymentMessage.PaymentMessageBuilder paymentMessageBuilder = PaymentMessage.builder()
-                .eventType(EventType.PAYMENT_PAID)
-                .eventTimestamp(now)
-                .namespace(payment.getNamespace())
-                .paymentId(payment.getRefundPaymentId())
-                .orderId(payment.getOrderId())
-                .userId(payment.getUserId())
-                .paymentPaidTimestamp(now)
-                .orderType(payment.getRefundType());
+        RefundMessage.RefundMessageBuilder refundMessageBuilder = RefundMessage.builder()
+                .eventType(EventType.REFUND_PAID)
+                .timestamp(now)
+                .namespace(refundPayment.getNamespace())
+                .refundId((refundPayment.getRefundId()))
+                .refundId(refundPayment.getRefundPaymentId())
+                .orderId(refundPayment.getOrderId())
+                .userId(refundPayment.getUserId());
 
-        PaymentMessage paymentMessage = paymentMessageBuilder.build();
+        RefundMessage refundMessage = refundMessageBuilder.build();
 
-        orderPaidWebHookAction(paymentMessage);
-
-        sendEventService.sendEventMessage(TopicName.PAYMENTS, paymentMessage);
-        log.debug("triggered event REFUND_PAID for refundId: " + payment.getRefundPaymentId());
+        refundPaidWebHookAction(refundMessage);
+        log.debug("triggered event REFUND_PAID for refundId: " + refundPayment.getRefundPaymentId());
     }
 
-    protected void orderPaidWebHookAction(PaymentMessage message) {
+    protected void refundPaidWebHookAction(RefundMessage message) {
         try {
-            sendNotificationService.sendPaymentMessageNotification(message);
+            sendNotificationService.sendRefundMessageNotification(message);
         } catch (Exception e) {
             log.error("webhookAction: failed action after receiving event, eventType: " + message.getEventType(), e);
         }
     }
 
-    public void triggerRefundPaymentFailedEvent(Payment payment) {
+    public void triggerRefundPaymentFailedEvent(RefundPayment refund) {
         PaymentMessage paymentMessage = PaymentMessage.builder()
```

Comment:
Use RefundMessage here

Reply:
I was working on this until I noticed the mention on the KYV-780 task that handling of failed refunds was to be implemented later, so I basically committed the little what I had refactored and dropped the rest.
Since these what you pointed out are small changes to make this FailedEvent -method usable, I can do these now.